Switch to .toml files for new Pkg3 Registry process

NHDaly opened the pull request:
- Add Project.toml, Manifest.toml
- Remove REQUIRE

I directly converted our version specifications from the REQUIRE file (ie kept them as their corresponding `>=` meaning), giving us:
https://github.com/NHDaly/Blink.jl/blob/2c6e94d392b13cf20c6a0f025d31e57b187c9776/Project.toml#L22-L26

But I think more likely than not, we probably want to switch these to be semver versions. But I guess we should reflect first on whether those versions are even up to date anymore? I _vaguely_ recall wanting to set some upper bound version requirements in the past and not being able to, but I don't think we have any such problems right now.
Any thoughts on what's best for this?

Thanks!

- [ ] This is a reminder that after we merge this and start the registration PR, we should ping @travigd to start a registration PR for [WebIO.jl](https://github.com/JuliaGizmos/WebIO.jl).

NHDaly commented:
Oh and also, @pfitzseb, I think I remember that you are in the anti-manifest file camp? I think I am in the opposed group as well, since it will hide breakages in upstream packages from our CI unless we keep it up-to-date. I think I vote for removing it, but i wanted to upload it here just for posterity. :)

travigd commented:
You'll need to change the TravisCI file.

```yaml
script:
  - $TESTCMD -e 'using Pkg; Pkg.build(); Pkg.test("Blink"; coverage=true)'
```

Manifest should be removed imo (it will just create git diff noise and serve no purpose).

travigd commented:
WebIO PR: https://github.com/JuliaRegistries/General/pull/1473

pfitzseb commented on Project.toml:
```diff
@@ -0,0 +1,32 @@
+[compat]
+JSExpr = "≥ 0.3"
+Lazy = "≥ 0.11.3"
+WebIO = "≥ 0.3"
```
Should probably just use `"0.3, 0.4, 0.5, 0.6, 0.7, 0.8"` here.

pfitzseb replied:
Hm, right, just `0.8` would probably be better. I really don't like not upper bounding deps, and we shouldn't claim compatibility with a (as of yet non-existant) WebIO v0.9.

NHDaly replied:
Thanks, I've tightened all the versions to be caret specifiers. I think this sounds good.

pfitzseb commented on Manifest.toml:
```diff
@@ -0,0 +1,286 @@
+# This file is machine-generated - editing it directly is not advised
```
Yeah, let's not have this in the repo, imho.

pfitzseb commented on Project.toml:
```diff
@@ -0,0 +1,32 @@
+WebSockets = "104b5d7c-a370-577a-8038-80a2059c5097"
+
+[compat]
+JSExpr = "≥ 0.3"
```
Should probably be `"0.5"`.

NHDaly replied:
Just to clarify, do you mean `"≥ 0.5"`, or literally just `"0.5"`?

pfitzseb replied:
I meant `"0.5"`, which is equivalent to `"^0.5"`, so we wouldn't claim compatibility with e.g. 0.6 (same below :)).

NHDaly replied:
Done, sounds good. Thanks!

pfitzseb commented on Project.toml:
```diff
@@ -0,0 +1,32 @@
+JSExpr = "≥ 0.3"
+Lazy = "≥ 0.11.3"
+WebIO = "≥ 0.3"
+WebSockets = "≥ 1.0.1"
```
Maybe `"1.5"`.

pfitzseb commented on Project.toml:
```diff
@@ -0,0 +1,32 @@
+Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
+MacroTools = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
+Mustache = "ffc61752-8dc7-55ee-8c37-f3e9cdd09e70"
+Mux = "a975b10e-0019-58db-a62f-e48ff68538c9"
```
Would probably be best to require Mux v0.7.

pfitzseb commented on Project.toml:
```diff
@@ -0,0 +1,32 @@
+name = "Blink"
+uuid = "ad839575-38b3-5650-b840-f874b8c74a25"
+authors = []
+version = "0.10.2"
```
Might make sense to directly jump to `0.11`, considering the rather big changes in regard to WebIO.

NHDaly commented:
> You'll need to change the TravisCI file.
> 
> ```yaml
> script:
>   - $TESTCMD -e 'using Pkg; Pkg.build(); Pkg.test("Blink"; coverage=true)'
> ```

Ah, thanks, good call! :) I've also changed the travis file for our docs build, which will hopefully work correctly as well.

> Manifest should be removed imo (it will just create git diff noise and serve no purpose).

👍 I removed the Manifest file as well. Thanks!

NHDaly commented:
@travigd  actually, unfortunately it looks like the latest WebIO `v0.8.6` has broken Blink.

Can that version be fixed and maybe you can bump the breaking changes to be `v0.9` so that we can keep an upper bound on `[v0.3 - v0.8]` and still work?
Here's the log showing the breakage:
```julia
(Blink) pkg> test Blink
     ...
   Testing Blink tests passed

(Blink) pkg> update
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
 Installed WebIO ─ v0.8.6
  Updating `~/.julia/dev/Blink/Project.toml`
  [0f1e0344] ↑ WebIO v0.8.5 ⇒ v0.8.6
  Updating `~/.julia/dev/Blink/Manifest.toml`
  [cd3eb016] ↑ HTTP v0.8.2 ⇒ v0.8.3
  [0f1e0344] ↑ WebIO v0.8.5 ⇒ v0.8.6
  Building WebIO → `~/.julia/packages/WebIO/9qsL8/deps/build.log`

(Blink) pkg> build
  Building MbedTLS → `~/.julia/packages/MbedTLS/X4xar/deps/build.log`
  Building WebIO ──→ `~/.julia/packages/WebIO/9qsL8/deps/build.log`
  Building Blink ──→ `~/.julia/dev/Blink/deps/build.log`
 Resolving package versions...

(Blink) pkg> test Blink
   Testing Blink
 Resolving package versions...
ERROR: LoadError: Javascript error      Error: Cannot find module 'debug'
```

NHDaly commented:
(and to clarify, i get the same issues even if I pin `HTTP` back to `v0.8.5`, so I think the issue is with WebIO):

```julia
(Blink) pkg> test Blink
   Testing Blink
 Resolving package versions...
ERROR: LoadError: Javascript error      Error: Cannot find module 'debug'
...
ERROR: Package Blink errored during testing

(Blink) pkg> status
Project Blink v0.11.0
    Status `~/.julia/dev/Blink/Project.toml`
...
  [cd3eb016] + HTTP v0.8.2 #6748814 (https://github.com/JuliaWeb/HTTP.jl.git) ⚲
...
  [0f1e0344] + WebIO v0.8.6
...
```

NHDaly commented:
Cool! Sorry this got dropped for two weeks.

I was hoping to finish it before I went on vacation all of last week, but didn't manage it.

Blink is now broken on master from the latest WebIO change; I've opened an issue about it here: https://github.com/JuliaGizmos/WebIO.jl/issues/317
Pending that fix, this PR should be good to go! :)

In the meantime I guess we could also change our compatability to more specifically upper-bound against `v0.8.6`?

NHDaly commented:
Okay, I've just gone ahead and downgraded `WebIO` to `"0.7"` for now. Assuming that seems reasonable, this should be good to merge once the CI build succeeds. :)

pfitzseb commented:
Alright, everything's green. Wanna merge + tag? :)
